client/extensions/woocommerce/state/data-layer/products/index.js

```diff
@@ -1,25 +1,51 @@
+/** @format */
```

Comment:
Just a note, I've been moving the new files like these to `state/sites/products/handlers.js` with the goal of eventually removing the `data-layer` directory.

Reply:
Why wouldn't we follow the core calypso pattern of [mirroring endpoint structure under the `data-layer` folder](https://github.com/Automattic/wp-calypso/tree/master/client/state/data-layer#file-structure)? This would keep all the code that calls endpoints in one location (I mean, eventually 🙂 ).

Reply:
I guess I see handlers like these as much more application-specific instead of infrastructure-specific. I think general purpose things like http requests and all should belong in another place, but I think handlers like these are much closer to the action and reducer code than the data layer itself.